src/openms/include/OpenMS/FORMAT/FileHandler.h

```diff
@@ -221,20 +222,23 @@ namespace OpenMS
         break;
       }
 
-      SourceFile src_file;
-      src_file.setNameOfFile(File::basename(filename));
-      src_file.setPathToFile(String("file:///") + File::path(filename));
-      // this is more complicated since the data formats allowed by mzML are very verbose.
-      // this is prone to changing CV's... our writer will fall back to a default if the name given here is invalid.
-      src_file.setFileType(FileTypes::typeToMZML(type));
-
-      if (compute_hash)
+      if (rewrite_source_file)
       {
-        src_file.setChecksum(computeFileHash_(filename), SourceFile::SHA1);
-      }
+        SourceFile src_file;
+        src_file.setNameOfFile(File::basename(filename));
+        src_file.setPathToFile(String("file:///") + File::path(filename));
+        // this is more complicated since the data formats allowed by mzML are very verbose.
+        // this is prone to changing CV's... our writer will fall back to a default if the name given here is invalid.
+        src_file.setFileType(FileTypes::typeToMZML(type));
+
+        if (compute_hash)
```

Comment:
`compute_hash` makes only sense when `rewrite_source_file` is set? maybe add a `PRECONDITION` that checks this, e.g. it should never happen that compute hash is true and rewrite is false